annotation-file-utilities/src/annotator/find/ASTPathCriterion.java

```diff
@@ -383,9 +383,8 @@ private Tree getNext(Tree actualNode, ASTPath astPath, int ix) {
                 } else if (astNode.childSelectorIs(ASTPath.BOUND)) {
                   return arg < 0 ? clazz.getExtendsClause()
                           : clazz.getImplementsClause().get(arg);
-                } else {
-                    return null;
                 }
+                return null;
```

Comment:
There's no final `else` -- only `else if` -- so there could be a fall-through.  (I believe the other cases won't ever get constructed, but only a whole-program analysis could figure that out.)